### PR TITLE
feat(publishing): allow assuming role for AWS CodeArtifact

### PR DESCRIPTION
**Pull request by AminFazlMondo**
Added ability to assume role prior to get authorization token from AWS CodeArtifact

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/node-package.ts [hunk 1]
```diff
@@ -307,6 +307,14 @@ export interface CodeArtifactOptions {
...
     */
   readonly secretAccessKeySecret?: string;
+
+  /**
+    * ARN of AWS role to be assumed prior to get authorization token from AWS CodeArtifact
+    * This property must be specified only when publishing to AWS CodeArtifact (`registry` contains AWS CodeArtifact URL).
+    *
+    * @default "AWS_SECRET_ACCESS_KEY"
+    */
+  readonly roleToAssume?: string;
```
Comment (eladb) [inline src/node-package.ts:317]: Should be under `CodeArtifactOptions`

**Review by eladb — CHANGES_REQUESTED** [review]
See comment

File: src/node-package.ts (hunk 1 above)
Comment (AminFazlMondo) [reply src/node-package.ts:317]: Does it not need to be here in order to use when defining projects?

Comment (eladb) [reply src/node-package.ts:317]: Users would define it via `codeArtifactOptions`

Comment (AminFazlMondo) [reply src/node-package.ts:317]: Do you think the following code change is not necessary?
https://github.com/projen/projen/blob/9135e799b7451fd1ea48cf7c33398a86d38a6e9d/src/node-project.ts#L630

**Review by eladb — APPROVED** [review]
LGTM

**Pull request merged**